client/my-sites/email/form/mailboxes/validators.ts

```diff
@@ -129,10 +130,15 @@ class MailboxNameValidator extends BaseValidator< string > {
 
 class PasswordResetEmailValidator extends BaseValidator< string > {
 	private readonly domainName: string;
+	private readonly setFieldIsVisible: MailboxForm< EmailProvider >[ 'setFieldIsVisible' ];
```

Comment:
The data flow in and around `NewMailBoxList` is admittedly flawed. My first instinct would have been to validate the `passwordResetEmail` field inside the `NewMailBoxList` component, but we can't do that since `hiddenFieldNames` is defined outside the component. The same limitation applies here – i.e., we can't change the field visibility from within a validator if we don't also modify `hiddenFieldNames` (and we definitely shouldn't attempt to do that from within here).

The ultimate solution would probably be to change how `hiddenFieldNames` is handled, but I'm OK with a cheaper solution for this PR. 

@chad1008, if we already have the `usePasswordResetEmailField` hook, do we really also need this `setFieldIsVisible` logic here? We already know that the default field value is OK because of the hook, and users can't edit the field without first making it visible, so an invisible invalid field seems very unlikely.

Reply:
You're right that we don't technically need it, as the hook will head off any issues. I opted to leave in this visibility logic, as well as setting `skipInvisibleFields` to `false` (one of your other comments below) to protect against either an edge case I'm not envisioning (I agree, an invalid invisible field with the new hook is very unlikely) or a potential future case where a new email setup flow is introduced, and the new hook isn't used for some reason.

It felt safer to make sure the form would validate a hidden field if the hook were ever not in play so we avoid the issue that led to this PR being necessary, but if I'm being overly cautious I'm happy to revert these parts.

> My first instinct would have been to validate the `passwordResetEmail` field inside the `NewMailBoxList` component, but we can't do that since hiddenFieldNames is defined outside the component.

Yep. I had a similar initial thought that I had to shoot down due to `hiddenFields`. Great minds...

Another advantage to doing it this way is that if we use the validator itself right off the bat, we can end up serving an error message right as the page loads, on a field we prefilled ourselves. This way we detect the error and clear the invalid value before the user sees an error they didn't even cause.